Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +353,391 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+                    kwargs[k] = v
+
+            integer_isa = translate_isa_string_to_enum(isa)
+            rbinfile, delete_after_use = return_file_with_rbins(binfile)
+            bin_refs = np.arange(1, maxbinref + 1)
+            bin_ref_perms = itertools.product(bin_refs, bin_refs)
+            dtype = np.dtype([(bytes_to_native_str(b'nRA'), np.int),
+                              (bytes_to_native_str(b'nDEC'), np.int),
+                              (bytes_to_native_str(b'avg_time'), np.float),
+                              (bytes_to_native_str(b'sigma_time'), np.float)])
+            all_runtimes = np.zeros(maxbinref ** 3, dtype=dtype)
+            all_runtimes[:] = np.inf
+
+            for ii, (nRA, nDEC) in enumerate(bin_ref_perms):
+                total_runtime = 0.0
+                total_sqr_runtime = 0.0
+
+                for _ in range(nrepeats):
+                    t0 = time.time()
+                    extn_results = DDtheta_mocks_extn(
+                        autocorr, nthreads, rbinfile,
+                        RA1, DEC1, RA2, DEC2,
+                        link_in_dec=link_in_dec,
+                        link_in_ra=link_in_ra,
+                        verbose=verbose,
+                        output_thetaavg=output_thetaavg,
+                        ra_refine_factor=1,
+                        dec_refine_factor=nDEC,
+                        max_cells_per_dim=max_cells_per_dim,
+                        isa=integer_isa)
+                    t1 = time.time()
+
+                    if extn_results is None:
+                        msg4 = "RuntimeError occurred with perms = ({0}, {1})". \
+                            format(nRA, nDEC)
+                        print(msg4)
+                        print("Continuing...")
+                        continue
+
+                    dt = (t1 - t0)
+                    total_runtime += dt
+                    total_sqr_runtime += dt * dt
+
+                avg_runtime = total_runtime / nrepeats
+
+                # variance = E(X^2) - E^2(X)
+                # disp = sqrt(variance)
+                runtime_disp = np.sqrt(total_sqr_runtime / nrepeats -
+                                       avg_runtime * avg_runtime)
+
+                all_runtimes[ii]['nRA'] = nRA
+                all_runtimes[ii]['nDEC'] = nDEC
+                all_runtimes[ii]['avg_time'] = avg_runtime
+                all_runtimes[ii]['sigma_time'] = runtime_disp
+
+            if delete_after_use:
+                import os
+                os.remove(rbinfile)
+
+            all_runtimes.sort(order=('avg_time', 'sigma_time'))
+            results = (all_runtimes[0]['nRA'],
+                       all_runtimes[0]['nDEC'])
+
+            optional_returns = return_runtimes
+            if not optional_returns:
+                ret = results
+            else:
+                ret = (results,)
+                if return_runtimes:
+                    ret += (all_runtimes,)
+
+            return ret
```

Comment:
You might be able to eliminate this entire `else` condition. Even if not, you can definitely reduce the number of repeated lines of code.

Reply:
I tried it, but it seems the `else` condition is necessary.

Reply:
@kakirastern Thanks again for your patience. I checked against the code-base, and if you set `nRA=1` (instead of `None`) - it should all be fine. `DDtheta_mocks` runs for me with this call (assuming the usual setup)

```python
In [11]: results = DDtheta_mocks(autocorr, nthreads, binfile,RA, DEC, output_thetaavg=True, link_in_dec=True, link_in_ra=False, ra_refine_factor=1, dec_refine_factor=3
    ...: , verbose=True)                                                                                                                                               
gridlink_mocks_theta_dec_double> expected_n = 2868 ngrid (declination) = 51 np=100000 Memory required = 3.37 MB. Time taken =   0.008 sec 
0%.........10%.........20%.........30%.........40%.........50%.........60%.........70%.........80%.........90%........100% done. Time taken =  1.882 secs


```


That should basically be it!

Reply:
No worries @manodeep I think that is what I have in the PR right now... Do I need to modify anything further besides setting `ra_refine_factor=1` for the `(link_in_dec=True, link_in_ra=False)` case?